janus.c

```diff
@@ -203,7 +206,14 @@ void janus_handle_signal(int signum)
 			JANUS_PRINT("Ok, leaving immediately...\n");
 			break;
 	}
+
+	/* Wake the main loop */
+	janus_mutex_lock(&stop_mutex);
```

Comment:
```stop_mutex``` is only ever used by the main loop, so I don't see how that may be happening? or you mean the signal might be interrupting the main loop as well and hence cause that issue?